erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -741,4 +741,22 @@ void WebRtcConnection::setTransport(std::shared_ptr<Transport> transport) {  //
   bundle_ = true;
 }
 
+std::pair<RTPExtensionsMap, RTPExtensionsMap> WebRtcConnection::getSourceExtensionMap() {
+  std::lock_guard<std::mutex> lk(extension_map_mutex);
+  auto result = std::make_pair(extension_processor_.getVideoExtensionMap(),
+                               extension_processor_.getAudioExtensionMap());
+  return result;
+}
+
+void WebRtcConnection::setSourceExtensionMap(std::shared_ptr<WebRtcConnection> source_wrtc) {
+  asyncTask([source_wrtc](std::shared_ptr<WebRtcConnection> this_connection){
+    auto source_map_future = source_wrtc->worker_->post_async_task([source_wrtc]{
+      return source_wrtc->getSourceExtensionMap();
+    });
+    auto map = source_map_future.get();
```

Comment:
Yep, doing "auto map = source_map_future.get()" waits for the result from another thread, so I think your option works better :smile: . And by making getSourceExtensionMap() private, we don't need extension_map_mutex any more.